spinta/utils/passwords.py

```diff
@@ -0,0 +1,52 @@
+import base64
+import hashlib
+import hmac
+import math
+import os
+import random
+
+
+def _bencode(value: bytes) -> str:
+    return base64.urlsafe_b64encode(value).decode().rstrip('=')
+
+
+def _bdecode(value: str) -> bytes:
+    value = value.ljust(math.ceil(len(value) / 4) * 4, '=')
+    return base64.urlsafe_b64decode(value)
+
+
+def crypt(password, iterations=None, salt=None):
+    if iterations is None:
+        iterations = random.randint(100_000, 999_999)
+    if salt is None:
+        salt = os.urandom(16)
+    elif isinstance(salt, str):
```

Comment:
should we check that `salt` is `bytes` , if won't be `None` or `str`?

Reply:
Python uses duck typing, that means, it does not matter if salt is bytes or not as long as it behaves like bytes.